pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -134,10 +140,12 @@ func addProxy(manifest *mf.Manifest) error {
 // transform mutates the passed manifest to one with common, component
 // and platform transformations applied
 func (r *Reconciler) transform(ctx context.Context, manifest *mf.Manifest, comp v1alpha1.TektonComponent) error {
-	//logger := logging.FromContext(ctx)
+	pipeline := comp.(*v1alpha1.TektonPipeline)
 	images := common.ToLowerCaseKeys(common.ImagesFromEnv(common.PipelinesImagePrefix))
 	instance := comp.(*v1alpha1.TektonPipeline)
 	extra := []mf.Transformer{
+		common.ReplaceConfigMapValues(featureFlag, pipeline.Spec.PipelineProperties),
+		common.AddOptionalConfigMapValues(configDefaults, pipeline.Spec.OptionalPipelineProperties),
```

Comment:
hi, could you tell us why we need 2 different transformers here.
From my prespective, at this point, all values can be consirered optional. That is if it is not set in the CR the default from the release.yaml (payload) will be already there in each of the 2 configMaps. so why don't we call `ReplaceConfigMapValues` 2 times but with 2 different configMap names

Reply:
for `config-defaults` cm there are not fields defined in configmap, all are commented so we loop n the `OptionalPipelineProperties` and if there is value defined for a field we add it in the configmap

for the other one, we loop on the config map fields and copy the value from CR, 
we set the default values in webhook and then copy them in transformer without any check.

Reply:
but now I think we can use `AddOptionalConfigMapValues` for both configmaps where we loop on struct fields and add in configmap